docs/topics/control-flow.md

```diff
@@ -50,71 +50,152 @@ val max = if (a > b) {
 If you're using `if` as an expression, for example, for returning its value or
 assigning it to a variable, the `else` branch is mandatory.
 
-## When expression
+## When expressions and statements
 
-`when` defines a conditional expression with multiple branches. It is similar to the `switch` statement in C-like languages.
-Its simple form looks like this.
+`when` is a conditional expression that executes code based on multiple possible values or conditions. It is
+similar to the `switch` statement in Java, C, and similar languages. For example:
 
 ```kotlin
+fun main() {
+    //sampleStart
+    val x = 2
+    when (x) {
+        1 -> print("x == 1")
+        2 -> print("x == 2")
+        else -> print("x is neither 1 nor 2")
```

Comment:
Let's add a code line reflecting the output, also as per our SG

Reply:
Added for other samples too.